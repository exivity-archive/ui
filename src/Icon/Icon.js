import PropTypes from 'prop-types'
import { styled, Inline } from 'reakit'
import { theme } from 'styled-tools'

import withEnumProps from '../withEnumProps'

export const Icon = styled(Inline)`  
  ${theme('Icon')};
`

Icon.propTypes = {
  ...Inline.propTypes,
  children: PropTypes.node,

  sup: PropTypes.node,
  sub: PropTypes.node,

  scale: PropTypes.number,
  xsmall: PropTypes.bool,
  small: PropTypes.bool,
  large: PropTypes.bool,
  xlarge: PropTypes.bool
}

Icon.defaultProps = {
  scale: 1
}

Icon.displayName = 'Icon'

export default withEnumProps(Icon, { scale: 'value' })
