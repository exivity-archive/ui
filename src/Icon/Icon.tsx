import React from 'react'
import styled from 'styled-components'
import { defaultStyledProps, fromTheme, StyledProps } from '../utils/styled'

export interface IconProps extends StyledProps {}

export const Icon = styled.span <IconProps>`
  font-size: ${fromTheme(theme => theme.global.baseSize)}px;
  height: 1em;
  line-height: 1em;
  text-transform: none;
`

Icon.defaultProps = defaultStyledProps
