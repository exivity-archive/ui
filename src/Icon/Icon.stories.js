import React from 'react'
import { Grid } from 'reakit'
import { storiesOf } from '@storybook/react'

import Icon from './Icon'

storiesOf('atoms|Icon', module)
  .add('default', () => <Icon>favorite</Icon>)
  .add('color', () => <Icon palette='primary'>favorite</Icon>)
  .add('scale', () => <Grid gridTemplateColumns='repeat(5, max-content)' gap={20} align>
    <Icon xsmall>favorite</Icon>
    <Icon small>favorite</Icon>
    <Icon>favorite</Icon>
    <Icon large>favorite</Icon>
    <Icon xlarge>favorite</Icon>
  </Grid>)
