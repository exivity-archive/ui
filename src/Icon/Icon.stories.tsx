import React from 'react'

// @ts-ignore
import { storiesOf } from '@storybook/react'

import { MdFavorite } from 'react-icons/md'
import Icon from './Icon'

storiesOf('atoms|Icon', module)
  .add('default', () => <Icon><MdFavorite /></Icon>)
