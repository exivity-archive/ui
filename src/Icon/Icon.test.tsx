/* eslint-disable no-undef */
import React from 'react'
import renderer from 'react-test-renderer'

import Icon from './Icon'

test('renders basic icon', () => {
  const button = renderer.create(<Icon/>)
  expect(button.toJSON()).toMatchSnapshot()
})
