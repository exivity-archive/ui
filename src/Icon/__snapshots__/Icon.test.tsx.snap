// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders basic icon 1`] = `
.c0 {
  height: 1em;
  line-height: 1em;
  text-transform: none;
}

<span
  className="c0"
/>
`;
